modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -834,6 +848,24 @@ ONNXCompiled::ONNXCompiled(const gapi::onnx::detail::ParamDesc &pp)
     out_data.resize(params.num_out);
 }
 
+ONNXCompiled::~ONNXCompiled() {
+    if(workload_type.has_value())
+    {
+        workload_type.value().get().removeListener(workloadId);
+    }
+}
+
+void ONNXCompiled::setWorkLoadType(const std::string &type) {
+    const char* keys[] = {"ep.dynamic.workload_type"};
+    const char* values[] = {type.c_str()};
+    this_session.SetEpDynamicOptions(keys, values, 1);
+}
+
+void ONNXCompiled::setWorkLoadRef(cv::gapi::wip::ov::WorkloadTypeRef workload) {
```

Comment:
I think this method should be named another way, It adds listeners, configures `workload_type` variable in `ONNXCompiled` class, but it actually doesn't set `workload_type` of ONNX to some value.

Reply:
I think it can be `configureWorkloadType`